tests/testthat/examples/compare.cpp

```diff
@@ -0,0 +1,20 @@
+template <typename real_t>
+real_t ldpois(real_t x, real_t lambda) {
+  return x * std::log(lambda) - lambda - std::lgamma(x + 1);
+}
+
+// [[odin.dust::compare_data(incidence = double)]]
```

Comment:
This is a default. Could it be `float` or even `int`? How would one set that?

Reply:
ah, it's not actually a default, it's a bit of a trick. The user provides a mapping of name => type here so you might have:

```c++
// [[odin.dust::compare_data(incidence = double)
// [[odin.dust::compare_data(N_tot = int)
```

The type will get used as the type in the generated struct and the template argument to `cpp11::as_cpp()`. That does complicate using floats here at the moment as we need to write a template specialisation for `cpp11::as_cpp<float>(SEXP)` but I think we could do that (we might also specially treat `real_t` here if that's not easy).

It's possible that if `as_cpp` is easily extensible that this could allow for compound types, and vectors might also work out of the box.

But I suspect that for our immediate use, doubles or ints will be enough to get started